lib/spack/spack/__init__.py

```diff
@@ -134,6 +134,9 @@
 misc_cache = FileCache(misc_cache_path)
 
 
+binary_cache_retrieved_specs = set()
```

Comment:
Is this caching necessary? It looks like Spack won't recognize any changes in the mirror after it downloads it once with this new caching. It might be nice to be able to pull down new binaries added to the mirror.

Reply:
This is in-memory caching so if the mirror changes in the middle of a spack command then it would not pick up the new changes. If the mirror changes between invocations then it will pick up the new specs.

That being said caching is definitely not required but avoids spidering all provided mirrors each time Spack checks to see if a binary version of the package is available (once for each spec in a given DAG).

Reply:
That's my brain fart, in memory caching is a good idea here.